bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedMetadataProvider.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.items;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.common.registry.AbstractManagedProvider;
+import org.eclipse.smarthome.core.storage.StorageService;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ManagedMetadataProvider} is an OSGi service, that allows to add or remove
+ * metadata for items at runtime. Persistence of added metadata is handled by
+ * a {@link StorageService}.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+@Component(immediate = true, service = { MetadataProvider.class, ManagedMetadataProvider.class })
+public class ManagedMetadataProvider extends AbstractManagedProvider<Metadata, MetadataKey, Metadata>
```

Comment:
Would it be possible to add `@NonNullByDefault`?

Reply:
same here